src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2450,7 +2450,21 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatement(SyntaxKind par
                     }
 
                     var incompleteMember = _syntaxFactory.IncompleteMember(attributes, modifiers.ToList(), type.IsMissing ? null : type);
-                    if (incompleteMember.ContainsDiagnostics)
+
+                    var containsErrorDiagnostic = false;
+                    foreach (var child in incompleteMember.EnumerateNodes())
```

Comment:
@jaredpar: 

> What happens though if the diagnostic is in a grandchild of the original node? We'd miss that here.

EnumerateNodes yields all nodes underneath it.

> Still suspicious that this is the correct approach. I dug through the parser and couldn't find any other places where we call EnumerateNodes

This might be one of the only places where that would get checked seeing as this is responsible for producing IncompleteMember definitions.

>or check DiagnosticSeverity.

It is likely that there are other bugs around this in the codebase.